### DISCUSSION TITLE
How to use the default DNS server, but through dnscrypt proxy in order to get the filtering benefits.

### DISCUSSION THREAD

**Discussion by spa5k** [DNSCrypt/dnscrypt-proxy#2117] [Q&A]
Same as title

**Comment by lifenjoiner** [comment]
https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Forwarding
It's an allowed list. And you should get your default DNS ready first.

**Comment by spa5k** [comment]
But this seems to be pretty verbose since i want to simply skip the server provided by dnscrypt-proxy

**Reply by lifenjoiner** [reply]
Try Generic top-level domain (gTLD). Some you visit most will help a lot.

**Comment by galpt** [comment]
> But this seems to be pretty verbose since i want to simply skip the server provided by dnscrypt-proxy

When you use `dnscrypt-proxy`, you're telling that you want `dnscrypt-proxy` to handle those requests instead of sending them directly to your router. You can either use _localhost (127.0.0.1)_ or your _assigned LAN IP address (192.168.x.x)_. Both will work just fine.

If I'm not wrong, using [Forwarding](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Forwarding) is not what you wanted because the `forwarding-rules.txt` file needs to be constantly updated.

You may want to use the [Static entries](https://github.com/DNSCrypt/dnscrypt-proxy/blob/38e87f9a7bbcf455c33cc7a2e43c82bb77f1ba8c/dnscrypt-proxy/example-dnscrypt-proxy.toml#L852) and forward the requests to your router (e.g. 192.168.1.1) so the `dnscrypt-proxy` will just act as a local filtering proxy instead of forwarding your requests to other public servers.
You can use the [DNS stamp calculator](https://dnscrypt.info/stamps/) to add your own servers.

Here's a quick example using 192.168.1.1 as the upstream server:
```
########################################
#            Static entries            #
########################################

## Optional, local, static list of additional servers
## Mostly useful for testing your own servers.

[static]

   [static.my-router]
     stamp = 'sdns://AAAAAAAAAAAACzE5Mi4xNjguMS4x'
```
Then, on [Global settings](https://github.com/DNSCrypt/dnscrypt-proxy/blob/38e87f9a7bbcf455c33cc7a2e43c82bb77f1ba8c/dnscrypt-proxy/example-dnscrypt-proxy.toml#L16), you can add `my-router` and test if that works like what you're expecting or not.
```
##################################
#         Global settings        #
##################################

## List of servers to use
##
## Servers from the "public-resolvers" source (see down below) can
## be viewed here: https://dnscrypt.info/public-servers
##
## The proxy will automatically pick working servers from this list.
## Note that the require_* filters do NOT apply when using this setting.
##
## By default, this list is empty and all registered servers matching the
## require_* filters will be used instead.
##
## Remove the leading # first to enable this; lines starting with # are ignored.

 server_names = ['my-router']
```
Personally, I would add `google` and/or `cloudflare` and use the [load balancing strategy](https://github.com/DNSCrypt/dnscrypt-proxy/blob/38e87f9a7bbcf455c33cc7a2e43c82bb77f1ba8c/dnscrypt-proxy/example-dnscrypt-proxy.toml#L142) to ensure high availability. But it's up to you.

Also, you may want to set `require_dnssec`, `require_nolog`, `require_nofilter` to `false` since most routers or your ISP's servers usually won't meet one of those criteria.

**Reply by lifenjoiner** [reply]
I'm afraid plain DNS is not supported.

**Reply by galpt** [reply]
> I'm afraid plain DNS is not supported.

Ahh, okay~
Guess @spa5k has to use the available upstream servers then.

**Reply by lifenjoiner** [reply]
Your suggestion is inspiring. Using [encrypted-dns-server](https://github.com/DNSCrypt/encrypted-dns-server) as the upstream will make it happen.
